iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/relational/tvf/VariationTableFunction.java

```diff
@@ -147,13 +234,20 @@ public void finish(
       outputWindow(properColumnBuilders, passThroughIndexBuilder);
     }
 
-    private void outputWindow(
+    protected void outputWindow(
         List<ColumnBuilder> properColumnBuilders, ColumnBuilder passThroughIndexBuilder) {
+      boolean increaseIndex = false;
       for (long i = currentStartIndex; i < curIndex; i++) {
+        if (!skipIndex.isEmpty() && i == skipIndex.peek()) {
+          // skip the index if it is in the skip queue
+          skipIndex.poll();
+          continue;
+        }
```

Comment:
we should also output those null line., like 1, null, 1, null, 2, null, 2, 3, null, 3, null,
first group contains 1, null, 1, null
secodn group contains 2, null, 2,
third group contains 3, null, 3, null